core/src/main/scala/kafka/tools/StorageTool.scala

```diff
@@ -59,10 +59,14 @@ object StorageTool extends Logging {
         case "format" =>
           val directories = configToLogDirectories(config.get)
           val clusterId = namespace.getString("cluster_id")
-          val metadataVersion = getMetadataVersion(namespace, Option(config.get.interBrokerProtocolVersionString))
+          val metadataVersion = getMetadataVersion(namespace,
+            Option(config.get.originals.get(KafkaConfig.InterBrokerProtocolVersionProp)).map(_.toString))
           if (!metadataVersion.isKRaftSupported) {
             throw new TerseFailure(s"Must specify a valid KRaft metadata version of at least 3.0.")
           }
+          if (!metadataVersion.isProduction()) {
+            throw new TerseFailure(s"Metadata version ${metadataVersion} is not ready for production use yet.")
+          }
```

Comment:
I created an internal configuration `unstable.metadata.versions.enable` to bypass this. Since it is internal and undocumented, it doesn't need a KIP.

(Also see the very similar `unstable.api.versions.enable`)